components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/dao/RegistrySAMLSSOServiceProviderDAOImpl.java

```diff
@@ -253,10 +254,22 @@ protected SAMLSSOServiceProviderDO resourceToObject(Resource resource) {
     }
 
     @Override
-    public boolean addServiceProvider(SAMLSSOServiceProviderDO serviceProviderDO, int tenantId)
-            throws IdentityException {
+    /**
 
-        Registry registry = getRegistry(tenantId);
+     * Add the service provider information to the registry.
+
+     * @param serviceProviderDO Service provider information object.
+
+     * @return True if addition successful.
+
+     * @throws IdentityException Error while persisting to the registry.
+
+     */
+
+    public boolean addServiceProvider(SAMLSSOServiceProviderDO serviceProviderDO, int tenantId) throws IdentityException {
+
+        registry = getRegistry(tenantId);
+//        Registry registry = getRegistry(tenantId);
```

Comment:
`registry = getRegistry(tenantId);` part should be removed.
`Registry registry = getRegistry(tenantId);` part should be uncommented.

Reply:
Resolved in [Introduce JDBC based persistence for SAML](https://github.com/wso2/carbon-identity-framework/pull/6241/commits/d1e67ce649f7a5d2e7e62f364ce93815f347e7b4)